src/vos/tests/vos_tests.c

```diff
@@ -137,53 +137,89 @@ main(int argc, char **argv)
 	}
 
 	gc = 0;
-	if (argc < 2) {
-		nr_failed = run_all_tests(0, false);
-	} else {
-		while ((opt = getopt_long(argc, argv, "apcdglnti:XA:h",
+	bool test_run = false;
+
+	while ((opt = getopt_long(argc, argv, "apcdglnti:XA:hf:e:",
+				  long_options, &index)) != -1) {
+		switch (opt) {
+		case 'e':
+#if CMOCKA_FILTER_SUPPORTED == 1 /** requires cmocka 1.1.5 */
+			cmocka_set_skip_filter(optarg);
+#else
+			D_PRINT("filter not enabled");
+#endif
+
+			break;
+		case 'f':
+#if CMOCKA_FILTER_SUPPORTED == 1 /** requires cmocka 1.1.5 */
+			cmocka_set_test_filter(optarg);
+				printf("Test filter: %s\n", optarg);
+#else
+			D_PRINT("filter not enabled");
+#endif
+			break;
+		default:
+			break;
+		}
+	}
+	index = 0;
+	optind = 0;
+
+	while ((opt = getopt_long(argc, argv, "apcdlnti:A:hf:e:",
```

Comment:
I think this is my fault (when I was merging my last patch with master), but the options strings don't match between the two while loops. Namely "X" and "g" aren't in the second one so those tests can't be run individually with command line args. I can fix it in an upcoming patch, but if you need to make other commits to this PR, might be a good opportunity to fix this as well :-).

Reply:
ah, let me fix it.

Reply:
fixed...this should solve the root problem that we have two lists to maintain